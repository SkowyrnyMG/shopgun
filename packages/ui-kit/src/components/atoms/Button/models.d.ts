import { KeyboardEvent } from 'react';

export namespace BUTTON {
	export enum COLOR {
		DARK = 'dark',
		LIGHT = 'light',
	}

	export enum SIZE {
		LG = 'large',
		MD = 'medium',
		SM = 'small',
		XS = 'extraSmall',
	}
}

export interface ButtonProps {
	onClick: (
		e:
			| React.MouseEvent<HTMLButtonElement, MouseEvent>
			| React.KeyboardEvent<HTMLButtonElement, KeyboardEvent>,
	) => void;
	color?: BUTTON.COLOR;
	size?: BUTTON.SIZE;
}
