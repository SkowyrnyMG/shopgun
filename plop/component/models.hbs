{{#insertOption typeOptions "interface"}}
export interface {{properCase name}}Props {

};
{{/insertOption}}

{{#insertOption typeOptions "namespace"}}
{{lowerCase namespaceOption}} namespace {{properCase name}}Type {

};
{{/insertOption}}