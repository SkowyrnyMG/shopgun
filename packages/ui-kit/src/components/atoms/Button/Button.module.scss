@import 'styles/main';

.wrapper {
	border: none;
	border-radius: 5px;
}

.dark {
	color: $c-gray-0;
	background-color: $c-blue-1;
}

.light {
	color: $c-white;
	background-color: $c-blue-2;
}

.large {
	padding: 20px 40px;
	@include font-size(20);
}

.medium {
	padding: 15px 30px;
	@include font-size(20);
}

.small {
	padding: 10px 20px;
	@include font-size(16);
}

.extraSmall {
	padding: 5px;
	@include font-size(14);
}
