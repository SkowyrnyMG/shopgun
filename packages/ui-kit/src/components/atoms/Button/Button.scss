.button {
  padding: 10px 20px;
  color: white;
  background: darkblue;
}
