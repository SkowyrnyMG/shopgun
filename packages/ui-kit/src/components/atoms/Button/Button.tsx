import React, { FC } from 'react';

import cn from 'classnames';

import button from './Button.module.scss';

interface ButtonProps {
  variant?: string;
}

export const Button: FC<ButtonProps> = ({ children, variant = 'blue' }) => {
  const buttonClasses = cn(button.wrapper, {
    [button[variant]]: variant,
  });

  return <button className={buttonClasses}>{children}</button>;
};
