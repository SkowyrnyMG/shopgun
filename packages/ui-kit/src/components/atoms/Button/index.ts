export { Button } from './button-base';
