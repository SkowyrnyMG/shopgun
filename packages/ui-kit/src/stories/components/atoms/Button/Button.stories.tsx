import React from 'react';

import { ComponentMeta, ComponentStory } from '@storybook/react';
import Button, { BUTTON } from 'components/atoms/Button';

export default {
	title: 'Components/Atoms/Button',
	component: Button,
	argTypes: {},
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
	children: 'Click me',
};

export const Light = Template.bind({});
Light.args = {
	children: 'Click me',
	color: BUTTON.COLOR.LIGHT,
};

export const Big = Template.bind({});
Big.args = {
	children: 'Click me',
	size: BUTTON.SIZE.LG,
};

export const Medium = Template.bind({});
Medium.args = {
	children: 'Click me',
	size: BUTTON.SIZE.MD,
};

export const Small = Template.bind({});
Small.args = {
	children: 'Click me',
	size: BUTTON.SIZE.SM,
};

export const ExtraSmall = Template.bind({});
ExtraSmall.args = {
	children: 'Click me',
	size: BUTTON.SIZE.XS,
};
