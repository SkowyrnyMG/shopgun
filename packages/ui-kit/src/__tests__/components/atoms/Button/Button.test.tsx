import React from 'react';

import Button from 'components/atoms/Button';
import { ButtonProps } from 'components/atoms/Button/models.d';
import { render, screen, userEvent } from 'testUtils';

const ButtonMock = {
	onClick: jest.fn(),
};

const props: ButtonProps = ButtonMock;

describe('Button - should pass all test cases', () => {
	test('the Button component renders correctly with required and optional props', () => {
		render(<Button {...props} />);

		const buttonNode = screen.getByTestId('test-button');

		// the component has been rendered
		expect(buttonNode).toBeInTheDocument();

		// generating a snapshot
		expect(buttonNode).toMatchSnapshot();
	});

	test('after user click on Button correct action should be triggered', () => {
		render(<Button {...props} />);

		const buttonNode = screen.getByTestId('test-button');

		userEvent.click(buttonNode);

		expect(ButtonMock.onClick).toBeCalled();
	});
});
