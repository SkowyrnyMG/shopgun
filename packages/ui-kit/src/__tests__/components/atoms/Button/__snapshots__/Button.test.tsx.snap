// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - should pass all test cases the Button component renders correctly with required and optional props 1`] = `
<button
  class="wrapper dark medium"
  data-testid="test-button"
/>
`;
